src/algebra/pointwise.lean

```diff
@@ -185,6 +185,14 @@ def pointwise_mul_comm_semiring [comm_monoid α] : comm_semiring (set α) :=
 
 local attribute [instance] pointwise_mul_semiring
 
+instance comm_monoid [comm_monoid α] : comm_monoid (set α) :=
+@comm_semiring.to_comm_monoid (set α) pointwise_mul_comm_semiring
+
+instance add_comm_monoid [add_comm_monoid α] : add_comm_monoid (set α) :=
+show @add_comm_monoid (additive (set (multiplicative α))), by apply_instance
+
+attribute [to_additive set.add_comm_monoid] set.comm_monoid
```

Comment:
The modern way seems to be to write `comm_monoid` after `add_comm_monoid` and mark the definition `@[to_additive]` without specifying the additive definition.

Reply:
I don't understand. I use `comm_monoid` in the definition of `add_comm_monoid` and if I don't then I'll have to duplicate code won't I?

Reply:
Oh yeah,  hadn't noticed that. It won't work on this occasion without more refactoring.

Reply:
What, if anything, do you want me to (try to) do?

Reply:
I don't really understand this file. Why is `def pointwise_mul_monoid [monoid α] : monoid (set α) := ...` on line 65 not an instance? Is it a bad instance somehow?

Reply:
Changed instances to defs

Reply:
@kbuzzard Johan points out on the original PR that these instances are "non-canonical", and I agree. This is not the only conceivable way to multiply two sets (to wit: intersection and symmetric difference form a ring), and having it be a global instance messes with other possible choices.

Reply:
Do I need to do anything else?